app/src/main/java/com/umairjavid/kombindsample/ui/main/MainActivity.kt

```diff
@@ -21,8 +24,12 @@ class MainActivity : KombindActivity<MainViewModel>() {
     private fun setupSimpleItemList() {
         simple_item_list.apply {
             layoutManager = LinearLayoutManager(this@MainActivity)
-            adapter = SimpleItemAdapter(viewModel.state.items, viewModel)
-                    .registerObserver(this@MainActivity)
+            adapter = object : Kombind_items_Adapter(viewModel.state.items, viewModel) {
```

Comment:
how is the name `Kombind_items_Adapter` determined? why not make it `KombindItemsAdapter`?

Reply:
the actual name of the class is taken from the annotated variable name.
So if you have @SimpleAdapter val someVariable: MutableLiveArrayList<String>
The generated class will be called Kombind_someVariable_Adapter.  For ref look at AdapterProcessor.kt Line 60